File: packages/mccomponents/libmccomponents/kernels/detector/EventModeMCA.cc
// -*- C++ -*-
//
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
//
//                                   Jiao Lin
//                      California Institute of Technology
//                        (C) 2007  All Rights Reserved
//
// {LicenseText}
//
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
//


#include "mccomponents/kernels/detector/EventModeMCA.h"


mccomponents::detector::EventModeMCA::EventModeMCA
(const char *outfilename, const index_t &npixels)
  : m_out( outfilename, std::ofstream::binary ),
    m_npixels( npixels )
{
}

mccomponents::detector::EventModeMCA::~EventModeMCA()
{
  
}

void mccomponents::detector::EventModeMCA::accept
( const channels_t & channels, double n )
{
  assert (channels.size()==3);
  if (channels[0] < 0 || channels[1]<0 || channels[2]<0) return;
  index_t pixelID = channels[0] * m_npixels + channels[1];
  m_out.write( (const char *)&pixelID, sizeof(index_t) );
  m_out.write( (const char *)(&channels[2]), sizeof(index_t) );
  m_out.write( (const char *)&n, sizeof(double) );
}



// version
// $Id$

// End of file 

File: packages/mccomponents/python/mccomponents/detector/elements/EventModeMCA.py
#!/usr/bin/env python
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                                   Jiao Lin
#                      California Institute of Technology
#                        (C) 2007 All Rights Reserved  
#
# {LicenseText}
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#


class EventModeMCA:

    def __init__(self, outfilename, detectorDims ):
        '''event mode multi channel analyzer

        outfilename: output data file name
        detectorDims: dimensions of detector system.
          For example, a detector system of 100 packs, 8 tubes per pack, 100 pixels per tube:
            detectorDims = 100, 8, 100
        '''

        self.outfilename = outfilename
        self.detectorDims = detectorDims
        return

    def identify(self, visitor):
        return visitor.onEventModeMCA(self)

    pass # end of EventModeMCA


# version
__id__ = "$Id$"

# End of file 
